pymc3/model.py

```diff
@@ -720,3 +720,50 @@ def all_continuous(vars):
 # theano stuff
 theano.config.warn.sum_div_dimshuffle_bug = False
 theano.config.compute_test_value = 'raise'
+
+
+# Replace theano as_op decorator with something friendly
+
+if sys.version_info < [3]:
+    def deterministic(f=None, **kwds):
+        itypes = kwds.get('itypes')
+        otypes = kwds.get('otypes') or [t.dvector]
+
+        # Wrapper infers types of inputs
+        @wraps(f)
+        def wrapper(*inputs, **kwinputs):
+            global itypes, otypes
+
+            if not itypes:
+                itypes = [arg.type for arg in inputs]
```

Comment:
Thanks I think it needs a bit of a refactor as tests are failing. However I think it's just a cosmetic change